lib/ae_mdw/node.ex

```diff
@@ -0,0 +1,147 @@
+defmodule AeMdw.Node do
+  @moduledoc """
+  Sample module to understand all of the functions the AwMdw.Node module
+  provides. Including it's specs as well.
+
+  Right now this module and its functions are defined using the
+  SmartGlobal library at runtime. The purpose of the module is to make
```

Comment:
Why do we need this?

Reply:
For two reasons:

* It's very hard to know what this module functions are, because they are built during runtime
* Dialyzer complains about function calls made to this module (because they are not available at compile-time)